src/main/kotlin/com/getkeepsafe/dexcount/DexMethodCountExtension.kt

```diff
@@ -16,94 +16,96 @@
 
 package com.getkeepsafe.dexcount
 
+import java.lang.IllegalArgumentException
+import java.util.Locale
+
 /**
  * Configuration properties for {@link DexMethodCountTask} instances.
  */
-class DexMethodCountExtension {
+open class DexMethodCountExtension {
     /**
      * The format of the method count output, either "list", "tree", "json", or "yaml".
      */
-    OutputFormat format = OutputFormat.LIST
+    var format: OutputFormat? = OutputFormat.LIST
+        set(value) {
+            if (value is OutputFormat) {
+                field = value
+            } else {
+                try {
+                    val formatName = "$value".toUpperCase(Locale.US)
+                    field = OutputFormat.valueOf(formatName)
+                } catch (ignored: IllegalArgumentException) {
+                    throw IllegalArgumentException("Unrecognized output format '$value'")
+                }
+            }
+        }
 
     /**
      * When true, individual classes will be include in the package list - otherwise, only packages
      * are included.
      */
-    boolean includeClasses
+    var includeClasses: Boolean? = false
```

Comment:
Why are fields here nullable - is it because type information isn't enforced in Gradle scripts?